core/object/script_language_extension.h

```diff
@@ -89,6 +90,12 @@ class ScriptExtension : public Script {
 
 		return class_doc;
 	}
+
+	virtual String get_class_icon_path() const override {
+		String ret;
+		GDVIRTUAL_REQUIRED_CALL(_get_class_icon_path, ret);
```

Comment:
This method is called for every node with unique script, so at worst you will get a one-time error spam when opening a scene.

Reply:
> So long as Godot does something sane when empty `String` is returned, and it isn't getting called often enough that the error spam would be a problem, I think this would be OK?

That would essentially be an "implicit default implementation" (relying on certain behavior by not implementing the function) -- would it not be better to provide a default right away? Or do you see a big risk that people forget to implement this?



> This method is called for every node with unique script, so at worst you will get a one-time error spam when opening a scene.

Still, this is quite bad UX and there's nothing a developer of an older extension can do against it (except recompile extensions under newer Godot versions, which makes them incompatible with _older_ ones). Godot has generally a bit a problem with errors appearing on the console, to a user it's often unclear if it's their fault or just a bug in the engine.

Reply:
Just to avoid unnecessary discussion, the only reason I made it a required call is because its cousin, `get_documentation` is. There is no deep reasoning for this and I can safely change it to a normal call. The implications weren't clear to me at the time.

I'll do the change once I'm back to work late next week.